atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleDdlTable.java

```diff
@@ -116,17 +116,21 @@ private void createOverflowTable() {
 
     private void putTableNameMapping(String fullTableName, String shortTableName) {
         if (config.useTableMapping()) {
-            try {
-                conns.get().insertOneUnregisteredQuery(
-                        "INSERT INTO " + AtlasDbConstants.ORACLE_NAME_MAPPING_TABLE
-                                + " (table_name, short_table_name) VALUES (?, ?)",
-                        fullTableName,
-                        shortTableName);
-            } catch (PalantirSqlException ex) {
-                if (ex.getMessage().contains(OracleErrorConstants.ORACLE_UNIQUE_CONSTRAINT_ERROR)) {
-                    dropTableInternal(fullTableName, shortTableName);
-                }
-                log.error(ex.getMessage(), ex);
+            insertTableMappingIgnoringPrimaryKeyViolation(fullTableName, shortTableName);
+        }
+    }
+
+    private void insertTableMappingIgnoringPrimaryKeyViolation(String fullTableName, String shortTableName) {
+        try {
+            conns.get().insertOneUnregisteredQuery(
+                    "INSERT INTO " + AtlasDbConstants.ORACLE_NAME_MAPPING_TABLE
+                            + " (table_name, short_table_name) VALUES (?, ?)",
+                    fullTableName,
+                    shortTableName);
+        } catch (PalantirSqlException ex) {
+            if (!ex.getMessage().toLowerCase().contains(AtlasDbConstants.NAME_MAPPING_PK_CONSTRAINT.toLowerCase())) {
```

Comment:
I'd suggest using `!StringUtils.containsIgnoreCase(ex.getMessage(), AtlasDbConstants.NAME_MAPPING_PK_CONSTRAINT)`. Maybe refactor this to a separate method `isPrimaryKeyViolation()`?